pkg/dashboard/apiserver/auth/oidc/utils.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package oidc
+
+import (
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"golang.org/x/oauth2"
+)
+
+func setCookie(c *gin.Context, token *oauth2.Token) {
```

Comment:
The `setCookie` has already been defined in `gcp` package, it's better to extract a public method instead of making a copy of it.